flight/mcu/lib/HistoryBuffer/HistoryBuffer.h

```diff
@@ -0,0 +1,144 @@
+                int currentIndex = (frontIndex + i)%size;
+                int pastIndex = (frontIndex + i+1)%size;
+
+                if(timestampArr[currentIndex]-timestampArr[pastIndex] != 0){
+                    firstDiff[i] = (arr[currentIndex] - arr[pastIndex])/((timestampArr[currentIndex]-timestampArr[pastIndex])*0.001);
+                }
+                else{
+                    firstDiff[i] = (arr[currentIndex] - arr[pastIndex])/(0.02);
+                }
+                
+            }
+
+            //we want to get acceleration from velocity, so we do the same thing but this time with the velocity array
+            //the new array with be length 49-1
+            float secondDiff[(size/2)-2];
+            for(unsigned i = 0; i < (size/2)-2; i++){
+                if(timestampArr[(frontIndex + i)%size]-timestampArr[(frontIndex + i+1)%size] != 0){
+                    secondDiff[i] = (firstDiff[i] - firstDiff[i+1])/((timestampArr[(frontIndex + i)%size]-timestampArr[(frontIndex + i+1)%size])*0.001);
+                }
+                else{
+                    secondDiff[i] = (firstDiff[i] - firstDiff[i+1])/(0.02);
+                }
+
+            }
+
+            //we then calculate the average of the acceleration array
+            float avg = 0.0;
+            for(unsigned i = 0; i < (size/2)-2; i++){
+                
+                avg += secondDiff[i];
+            }
+            return(avg/float(((size/2)-2))); 
+        }
+
+        float getPastSecondDerivativeAverage(){
+            //same as current second derivative average, except we use indexes 50-99
+            float firstDiff[(size/2)-1];
+            for(unsigned i = size/2; i < size-1; i++){
+                int currentIndex = (frontIndex + i)%size;
+                int pastIndex = (frontIndex + i+1)%size;
+                if(timestampArr[currentIndex]-timestampArr[pastIndex] != 0){
+                    firstDiff[i-(size/2)] = (arr[currentIndex] - arr[pastIndex])/((timestampArr[currentIndex]-timestampArr[pastIndex])*0.001);
+                }
+                else{
+                    firstDiff[i-(size/2)] = (arr[currentIndex] - arr[pastIndex])/(0.02);
+                }
+            }
+
+            float secondDiff[(size/2)-2];
+            for(unsigned i = 0; i < (size/2)-2; i++){
+                if(timestampArr[(frontIndex + i)%size]-timestampArr[(frontIndex + i+1)%size] != 0){
+                    secondDiff[i] = (firstDiff[i] - firstDiff[i+1])/((timestampArr[(frontIndex + i)%size]-timestampArr[(frontIndex + i+1)%size])*0.001);
+                }
+                else{
+                    secondDiff[i] = (firstDiff[i] - firstDiff[i+1])/(0.02);
+                }
+            }
+    
+            float avg = 0.0;
+            for(unsigned i = 0; i < (size/2)-2; i++){
+                avg += secondDiff[i];
+            }
+            return(avg/float(((size/2)-2))); 
+        }
+};
+#endif
```

Comment:
If you move the methods, HistoryBuffer is just a FifoBuffer. We can merge them together if you merge the FifoBuffer PR.